playwright/src/test/java/com/microsoft/playwright/TestElementHandleQuerySelector.java

```diff
@@ -0,0 +1,101 @@
+package com.microsoft.playwright;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.microsoft.playwright.Page.EventType.POPUP;
+import static com.microsoft.playwright.Page.LoadState.DOMCONTENTLOADED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TestElementHandleQuerySelector extends TestBase {
+  @Test
+  void shouldQueryExistingElement() {
+    page.navigate(server.PREFIX + "/playground.html");
+    page.setContent("<html><body><div class=\"second\"><div class=\"inner\">A</div></div></body></html>");
+    ElementHandle html = page.querySelector("html");
+    ElementHandle second = html.querySelector(".second");
+    ElementHandle inner = second.querySelector(".inner");
+    String content = inner.textContent();
+    String content2 = (String) page.evaluate("e => e.textContent", inner);
+    assertEquals(content, content2);
+    assertEquals(content, "A");
+  }
+
+  @Test
+  void shouldReturnNullForNonExistingElement() {
+    page.setContent("<html><body><div class=\"second\"><div class=\"inner\">B</div></div></body></html>");
+    ElementHandle html = page.querySelector("html");
+    ElementHandle second = html.querySelector(".third");
+    assertEquals(second, null);
```

Comment:
nit: expected value goes first `assertEquals(null, second)`, in this case you can use just `assertNull(second)`